core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/listener/DAnnotationChangeTrigger.java

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2021 THALES GLOBAL SERVICES. All rights reserved. 
+ *******************************************************************************/
+package org.polarsys.capella.core.sirius.ui.listener;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.emf.common.command.Command;
+import org.eclipse.emf.common.notify.Notification;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.sirius.business.api.dialect.command.RefreshRepresentationsCommand;
+import org.eclipse.sirius.business.api.session.ModelChangeTrigger;
+import org.eclipse.sirius.business.api.session.Session;
+import org.eclipse.sirius.ext.base.Option;
+import org.eclipse.sirius.ext.base.Options;
+import org.eclipse.sirius.viewpoint.DRepresentation;
+import org.eclipse.sirius.viewpoint.DRepresentationDescriptor;
+
+import com.google.common.collect.Sets;
+
+/**
+ * This ModelChangeTrigger looks for the parent representations of the notifiers and creates a
+ * {@link RefreshRepresentationsCommand} for them.
+ * 
+ * @author <a href="mailto:steve.monnier@obeo.fr">Steve Monnier</a>
+ */
+public class DAnnotationChangeTrigger implements ModelChangeTrigger {
+
+    private Session session;
+
+    public DAnnotationChangeTrigger(Session session) {
+        super();
+        this.session = session;
+    }
+
+    @Override
+    public Option<Command> localChangesAboutToCommit(Collection<Notification> notifications) {
+        LinkedHashSet<DRepresentation> representationsToRefresh = Sets.newLinkedHashSet();
+        for (Notification notification : notifications) {
+            if (notification.getNotifier() instanceof EObject) {
+                Option<DRepresentation> representationOption = getParentRepresentation((EObject) notification.getNotifier());
+                if (representationOption.some()) {
+                    representationsToRefresh.add(representationOption.get());
+                }
+            }
+        }
+        return Options.newSome(new RefreshRepresentationsCommand(session.getTransactionalEditingDomain(), new NullProgressMonitor(), representationsToRefresh));
+    }
+
+    private Option<DRepresentation> getParentRepresentation(EObject eObject) {
+        EObject current = eObject;
+        while (current != null) {
+            if (current instanceof DRepresentationDescriptor) {
+                current = ((DRepresentationDescriptor) current).getRepresentation();
```

Comment:
If we change the contextual elements of an unloaded representation, i think this will load it.
shall return empty option if not loaded i think.

Reply:
Good point, I fixed it